src/averageAges.js

```diff
@@ -55,7 +77,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => {
+    const isChild = people.some(human => human.name === person.mother);
+
+    return onlyWithSon ? (person.sex === 'm' && isChild) : isChild;
+  });
+
+  const diffAges = children.map(child => {
+    const mother = people.find(human => human.name === child.mother);
+
+    return child.born - mother.born;
+  });
+
+  return diffAges.reduce((a, b) => a + b, 0) / diffAges.length;
```

Comment:
Here also you can use `calcAvg` function like you have already used

Reply:
HI @amitpatiljc 
function calculateAverageAgeDiff returns arrays with only numbers
functions calculateMenAverageAge, calculateWomenAverageAge, returns object with filtered elements
The only solution that I see is simplify objects from calculateMenAverageAge and calculateWomenAverageAge (with map) and then calcAvg might be use for all cases